source/ShipyardPanel.cpp

```diff
@@ -255,16 +257,17 @@ void ShipyardPanel::Sell()
 		else
 		{
 			while(count-- > 1)
-				message += ",\n" + (*it++)->Name();
+				message += ",\n" + font.TruncateFront((*it++)->Name(), DIALOG_MAX_WIDTH - 33);
 			message += ",\nand ";
 		}
 		message += (*it)->Name();
 	}
 	else
 	{
 		auto it = playerShips.begin();
-		for(int i = 0; i < MAX_LIST - 1; ++i)
-			message += (*it++)->Name() + ",\n";
+		message += (*it++)->Name() + ",\n";
```

Comment:
It was indeed intentional, because the first line doesn't start with the ship's name and thus has different wrapping behavior than all the other lines.